src/main/scala/codacy/plugins/test/ITest.scala

```diff
@@ -70,12 +49,50 @@ trait ITest extends LogSupport {
                  prefix = "",
                  needsCompilation = false,
                  hasConfigFile = true,
-                 isClientSide = false,
+                 standalone = false,
                  hasUIConfiguration = true,
                  isDefault = true,
                  configFilenames = Set.empty)
   }
 
+  protected def createFullToolSpec(toolSpec: ToolSpec,
+                                   dockerToolDocumentation: DockerToolDocumentation): FullToolSpec = {
+    val patterns =
+      dockerToolDocumentation.toolSpecification
```

Comment:
is the .toolSpecification the same as toolSpec? For me seems a bit strange that this method receives a toolSpec and a dockerToolDocumentation but then all the code is transformations only relying on dockerToolDocumentation.

Also assume that some if you replace some .map with .flatMap we may avoid some flattens / getOrElse? But we can take a look together